### PR TITLE
Make `jpm sign` exit 1 or 0 as expected

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla-jetpack/jpm/issues/381

### CODE REVIEW

File: lib/sign.js [hunk 1]
```diff
@@ -26,8 +31,11 @@ function sign (manifest, program, options) {
 
   }).then(function(result) {
     console.log(result.success ? 'SUCCESS' : 'FAIL');
+    config.systemProcess.exit(result.sucess ? 0 : 1);
```
Comment (mstriemer) [inline lib/sign.js]: `result.sucess` ~> `result.success`

File: test/unit/test.sign.js [hunk 2]
```diff
@@ -593,6 +594,92 @@ describe('amoClient.PseudoProgress', function() {
...
+    runSignCmd().then(function() {
+      expect(mockProcessExit.call[0]).to.be.equal(0);
+      done();
+    }).catch(done);
+  });
+
+  it('should exit 1 on signing failure', function(done) {
+    runSignCmd({
+      StubAMOClient: makeAMOClientStub({
+        result: {sucess: false},
```
Comment (mstriemer) [inline test/unit/test.sign.js]: type of `success` happens here too.

**Comment by mstriemer** [comment]
There seems to be a lot of setup code to test one function. I'd rather see a function that just gets called directly with different values of `response.success`. Since you added the `describe('sign', ...)` block then maybe this setup code will eventually be required but it looks out of place to what is currently being tested.

r+wc

File: test/unit/test.sign.js (hunk 2 above)
Comment (kumar303) [inline test/unit/test.sign.js]: d'oh! auto-completion fail

Comment (kumar303) [inline test/unit/test.sign.js]: huh, I tested it on the command line too (maybe I only checked errors).

**Comment by kumar303** [comment]
> There seems to be a lot of setup code to test one function. 

Agreed. It's a high level function that integrates all other components to run the complete shell command. These are always tricky to test. I made it a little more like the `xpi` command test which let me remove at least one of the stubs.

**Pull request merged**